### ISSUE TITLE
Not getting error response when an exception is throw

### ISSUE THREAD

**Issue by jameswagoner** [dingo/api#383]
Route::api(['version' => 'v1', 'prefix' => 'api'], function()
    {
        Route::get('error', function()
        {
            try {
                $divide = 100/0;
            } catch(Exception $e) {
                throw new \Symfony\Component\HttpKernel\Exception\HttpException($e->getMessage);
            }
        });
    }

I get a 500 response but the body is empty.

**Comment by tymondesigns** [comment]
looks like the first param is the status code and the message is second http://api.symfony.com/2.7/Symfony/Component/HttpKernel/Exception/HttpException.html

**Comment by jameswagoner** [comment]
Updated with a 500 then my message; no dice.

I tried to get as generic as I could with these and the HttpException is the generic I could go. I am aiming for UnauthorizedHttpException for use with my JWT auth, but ran into this issue.

Only way I am getting the expected json response is with `API::response()->errorUnauthorized('Not Authroized')` which isn't so bad but really like the idea of throwing an exception when I need to an the API handling it automatically

**Comment by tymondesigns** [comment]
maybe try throwing `Symfony\Component\HttpKernel\Exception\UnauthorizedHttpException` instead.

Other than that not sure on the face of it. Maybe related to #382

**Comment by jameswagoner** [comment]
Don't think this is the same. TBH, I might not be doing JWT right, learning as I am going. This is what I am after

    Route::api(['version' => 'v1', 'prefix' => 'api'], function()
    {
        Route::post('login', function()
        {
            $auth = Sentinel::stateless(Input::only(['user', 'password']));

            if (!$auth) throw new \Symfony\Component\HttpKernel\Exception\UnauthorizedHttpException('not sure what a challenge string should be');

            return JWTAuth::fromUser($auth);
        });
    }

I post with incorrect user/password and only get an empty 500 response.

**Comment by jameswagoner** [comment]
Think I have an underlying issue; I started to protect my endpoints with `'protected' => true` and getting that same empty 500. I am sending both Accept and Authorization headers.

I am stumped. Is there anyway that this package could be partially configured?

**Comment by jameswagoner** [comment]
I discovered https://github.com/dingo/api/issues/369 and applied the fix there but didnt help anything with my empy 500 responses. 

Any thoughts?

**Comment by jameswagoner** [comment]
Okay, I knew I had to have misconfigured something....

I was enticed by `API::error()` and had set it to return `Response::json()` which I have known to break Dingo API but not before I added that listener.

Problem solved

**Comment by dtirer** [comment]
@jameswagoner Hi James, i'm actually having the same issue, but i'm not sure what you're referring to with API:error().  Could you be more specific?

**Comment by james-daddies** [comment]
`API::error()` is an function that gets called on error much like `App::error()`. I was using `Response::json()` to format my error messages which conflicts with Dingo. Dingo will already return json since that is what I specified in my Accept headers.

I am running things differently now TBH. I have changed my API to always return a 200 response with a success or failure value in addition to the data I return.
